ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```
@@ -155,6 +165,116 @@ static os_log_type_t KextLogLevelAsOSLogType(KextLog_Level level)
+                   os_log(s_daemonLogger, "Unable to run stat on panic at %s errno=%d strerror=%s\n", fullFileName.c_str(), errno, strerror(errno));
+                   continue;
+               }
+
+               if (panicLogTime >= lastRunTimestamp)
+               {
+                  // Check if 'org.vfsforgit.PrjFSKext(' occurs in the log
+                  // If it does, then PrjFSKext may have caused the panic and we should log an error
+                  FILE* fp = fopen(fullFileName.c_str(), "r");
+                  if (fp)
+                  {
+                      char* line = nullptr;
+                      size_t len = 0;
+                      ssize_t read;
+                      while ((read = getline(&line, &len, fp)) != -1)
+                      {
+                          if (strstr(line, "org.vfsforgit.PrjFSKext(") != nullptr)
+                          {
+                              ostringstream errorMessage;
+                              errorMessage << "Found panic at " << fullFileName.c_str() << "\n";
+                              LogDaemonError(errorMessage.str());
+                              break;
+                          }
+                      }
+                      
+                      fclose(fp);
+                   
+                      if (line)
+                      {
+                          free(line);
+                      }
+                  }
+              }
+           }
+        }
+    }
+    closedir(dr);
+
+    // Update timestamp file
+    if (!DoesPathExist(PanicLogTimestampDirectory))
+    {
+        mkdir(PanicLogTimestampDirectory, 0755);
+    }
+    
+    FILE* fp = fopen(PanicLogTimestampFile, "a");
+    if (fp)
+    {
+       fprintf(fp, "Completed Panic Sweep for :%ld\n", lastRunTimestamp);
+       fclose(fp);
+    }
+    else
+    {
+        os_log(s_daemonLogger, "Unable to save panic timestamp file %s errno=%d strerror=%s\n", PanicLogTimestampDirectory, errno, strerror(errno));
```

Typo? `PanicLogTimestampFile` rather than `PanicLogTimestampDirectory`.

---

Exellent catch here!